File: packages/php/src/GregorianDate.php
<?php

namespace UmmAlQura;

class GregorianDate
{
    public function __construct(
        private int $year,
        private int $month,
        private int $day
    ) {}

    public function getYear(): int { return $this->year; }
    public function getMonth(): int { return $this->month; }
    public function getDay(): int { return $this->day; }
    public function getCalendar(): string { return 'gregorian'; }

    public function toArray(): array
    {
        return [
            'year' => $this->year,
            'month' => $this->month,
            'day' => $this->day,
            'calendar' => $this->getCalendar(),
        ];
    }
}

File: packages/java/src/main/java/com/ummalqura/GregorianDate.java
package com.ummalqura;

/**
 * Gregorian date representation
 */
public class GregorianDate {
    private final int year;
    private final int month;
    private final int day;

    public GregorianDate(int year, int month, int day) {
        this.year = year;
        this.month = month;
        this.day = day;
    }

    public int getYear() { return year; }
    public int getMonth() { return month; }
    public int getDay() { return day; }
    public String getCalendar() { return "gregorian"; }

    @Override
    public String toString() {
        return String.format("%d-%02d-%02d", year, month, day);
    }
}
